src/components/table/index.js

```diff
@@ -0,0 +1,193 @@
+/* eslint-disable react/jsx-props-no-spreading */
+/* eslint-disable no-nested-ternary */
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+
+import {
+  DefaultTable,
+  TableContainer,
+  Tr,
+  Td,
+  Th,
+  ArrowIcon,
+  HeaderContainer,
+  HeaderText,
+} from './components';
+
+const Table = ({
+  dataSource,
+  colNames,
+  headerColor,
+  variant,
+  sortIcon,
+  stripped,
+  hoverable,
+  bordered,
+  ...rest
+}) => {
+  const [obtaineddataSource, setobtaineddataSource] = useState(dataSource);
+  const [colSortIndex, setSortIndex] = useState([]);
+
+  const sortDataAscending = (key, index) => {
```

Comment:
Perhaps sort could be a single function that takes a param which points if the sorting is ascending/descending. I can see that both are almost the same except for the keys comparison part, which can easily be handled by the prop that says if the sort is ascending or not.

Reply:
okay bro